### ISSUE TITLE
Illegal inode value recording by syscheck

### ISSUE THREAD

**Issue by mrtc0** [wazuh/wazuh#4886]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 3.12.1 | syscheck | Manager| Packages | Ubuntu 18.04 |

I don't know if this is a syscheck bug, but there was an alert in alerts.json with a weird inode in it

<summary>alert.json</summary>
<details>

```
{
   "timestamp":"2020-04-10T01:30:50.102+0000",
   "rule":{
      "level":7,
      "description":"Integrity checksum changed.",
      "id":"550",
      "firedtimes":7872,
      "mail":false,
      "groups":[
         "ossec",
         "syscheck"
      ],
      "pci_dss":[
         "11.5"
      ],
      "gpg13":[
         "4.11"
      ],
      "gdpr":[
         "II_5.1.f"
      ],
      "hipaa":[
         "164.312.c.1",
         "164.312.c.2"
      ],
      "nist_800_53":[
         "SI.7"
      ]
   },
   "agent":{
      "id":"****",
      "name":"****",
      "ip":"****"
   },
   "manager":{
      "name":"wazuh-manager"
   },
   "id":"1586482250.8931422",
   "cluster":{
      "name":"wazuh",
      "node":"wazuh-worker-2"
   },
   "full_log":"File '/path/to/file' modified\nMode: scheduled\nChanged attributes: inode\nOld inode was: '-9223372036854775808', now it is '2311061769'\n",
   "syscheck":{
      "path":"/path/to/file",
      "size_after":"2",
      "perm_after":"rw-------",
      "uid_after":"0",
      "gid_after":"0",
      "md5_after":"****",
      "sha1_after":"****",
      "sha256_after":"****",
      "uname_after":"root",
      "gname_after":"root",
      "mtime_after":"2019-11-15T04:03:43",
      "inode_before":-9.2233720368547758e+18,
      "inode_after":2311061769,
      "changed_attributes":[
         "inode"
      ],
      "event":"modified"
   },
   "decoder":{
      "name":"syscheck_integrity_changed"
   },
   "location":"syscheck"
}
```
</details>

It looks like `inode_before` is an integer value overflowing.  
This file obviously doesn't have a negative inode value.  
When this value is checked directly in the database, it is equal to the value of `INT_MAX` , as follows

```
sqlite> SELECT inode FROM fim_entry WHERE file = "/path/to/file";
2147483647
```

Also, there were many data with an inode of `2147483647`.

```
sqlite> SELECT COUNT(*) FROM fim_entry WHERE inode = 2147483647;
686
```

I'm not sure where this is a bug, but it seems to me that [this code](https://github.com/wazuh/wazuh/blob/2ccb1f40b2d123933964bfd3763e7beb60c29bc9/src/wazuh_db/wdb_fim.c#L538) , for example, is storing the `inode` in `int` instead of `int64` .

How can I help you to fix this bug? Thank you.

**Comment by Molter73** [comment]
Hi @mrtc0 

I have been studying this issue and I believe you might be right as to where the problem is, I would need your help in order to narrow it down a little bit more.

I need to know if this problem is specific to the manager or agents, and it would be really useful to get the output on the `ossec.log` with syscheck running on debug level 2, specifically the messages that read `DEBUG: (6321): Sending FIM event:...` in order to see if the inode is wrong when leaving syscheck and `DEBUG: (6336): Fim inode entries: xxxx path count: yyyy`.

**Comment by bah07** [comment]
Hi @mrtc0 
This issue is solved in [#4914](https://github.com/wazuh/wazuh/pull/4914)

Thanks for the report!
Regards.

**Comment by mrtc0** [comment]
Thanks for the fix !
